test/publish.test.js

```diff
@@ -19,25 +34,75 @@ test.afterEach.always(() => {
 });
 
 test.serial('Publish a release', async t => {
-  const owner = 'test_user';
-  const repo = 'test_repo';
-  const env = {GITLAB_TOKEN: 'gitlab_token'};
-  const pluginConfig = {};
-  const nextRelease = {gitHead: '123', gitTag: '@scope/v1.0.0', notes: 'Test release note body'};
-  const options = {repositoryUrl: `https://gitlab.com/${owner}/${repo}.git`};
-  const encodedRepoId = encodeURIComponent(`${owner}/${repo}`);
-  const encodedGitTag = encodeURIComponent(nextRelease.gitTag);
+  const gitlab = authenticate(env)
+    .post(releaseUrl, releaseBody)
+    .reply(200);
+
+  const result = await publish(pluginConfig, {env, options, nextRelease, logger: t.context.logger});
+
+  t.is(result.url, `https://gitlab.com/${encodedRepoId}/tags/${encodedGitTag}`);
+  t.deepEqual(t.context.log.args[0], ['Published GitLab release: %s', nextRelease.gitTag]);
+  t.true(gitlab.isDone());
+});
 
+test.serial('Publish a release with assets', async t => {
+  const cwd = 'test/fixtures/files';
+  const uploaded = {markdown: '[file.css](/uploads/file.css)', url: '/uploads/file.css', alt: 'file.css'};
+  const notes = `${nextRelease.notes}\n\n#### Assets\n\n* ${uploaded.markdown}`;
+  const assets = [['**', '!**/*.txt', '!.dotfile']];
   const gitlab = authenticate(env)
-    .post(`/projects/${encodedRepoId}/repository/tags/${encodedGitTag}/release`, {
-      tag_name: nextRelease.gitTag,
-      description: nextRelease.notes,
+    .post(releaseUrl, {...releaseBody, description: notes})
+    .reply(200);
+  const gitlabUpload = authenticate(env)
+    .post(`/projects/${encodedRepoId}/uploads`, body => {
```

Comment:
You can use a regex here rather than a function:
```
.post(`/projects/${encodedRepoId}/uploads`, /filename="file.css"/gm)
``

Reply:
done